Lib/test/test_uuid.py

```diff
@@ -340,6 +340,46 @@ def test_uuid1(self):
         equal(((u.clock_seq_hi_variant & 0x3f) << 8) |
                          u.clock_seq_low, 0x3fff)
 
+    @unittest.skipUnless(importable('ctypes'), 'requires ctypes')
+    @unittest.skipUnless(uuid._uuid_generate_time.restype is not None,
+                         'requires uuid_generate_time_safe(3)')
+    def test_uuid1_safe(self):
+        u = uuid.uuid1()
+        # uuid_generate_time_safe() may return 0 or -1 but what it returns is
+        # dependent on the underlying platform support.  At least it cannot be
+        # unknown (unless I suppose the platform is buggy).
+        self.assertNotEqual(u.is_safe, uuid.SafeUUID.unknown)
+
+    @unittest.skipUnless(importable('ctypes'), 'requires ctypes')
```

Comment:
Just a suggestion: Using `self.subTest()` would make the following four tests much shorter and IMO easier to understand.